File: backend/app/ecommerce/services/webhook/src/main/java/com/congthanh/webhook/model/entity/AbstractAuditEntity.java
package com.congthanh.webhook.model.entity;

import com.congthanh.webhook.listener.CustomAuditingEntityListener;
import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.LastModifiedBy;

import java.time.Instant;

@MappedSuperclass
@Data
@EntityListeners(CustomAuditingEntityListener.class)
public abstract class AbstractAuditEntity {

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, nullable = false)
    private Instant createdAt;

    @CreatedBy
    @Column(name = "created_by")
    private String createdBy;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;

    @LastModifiedBy
    @Column(name = "updated_by")
    private String updatedBy;

}

File: backend/app/management/core-services/auth-service/src/modules/auth/entities/abstract-audit.entity.ts
import { Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { AuditEntity } from '../../../common/interface/audit-entity.interface';

export abstract class AbstractAuditEntity implements AuditEntity {
  @CreateDateColumn({ name: 'created_at', update: false, default: () => 'CURRENT_TIMESTAMP', type: 'time without time zone' })
  createdAt: Date;

  @Column({ name: 'created_by', nullable: false, update: false })
  createdBy: string;

  @UpdateDateColumn({ name: 'updated_at', nullable: false, type: 'time without time zone' })
  updatedAt: Date;

  @Column({ name: 'updated_by', nullable: false })
  updatedBy: string;
}
